### PR TITLE
Fix #240,  Add Instruction on How To Use Workflows in Contributing.md

### LINKED ISSUE #240: Add Instructions on How to Use Workflows in Contributing.md
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I reviewed the [cFS README.md file](https://github.com/nasa/cFS/blob/main/README.md) to see if the feature is in the major future work.
* [x] I performed a cursory search to see if the feature request is relevant, not redundant, nor in conflict with other tickets.

**Is your feature request related to a problem? Please describe.**
Users may not know how to use our GitHub Actions Workflows. 

**Describe the solution you'd like**
Add instructions on how to use our Github Actions Workflows under Writing High Quality Code. Instructions includes where workflows can be found, how to analyze or interpret results, how to run and configure workflows. Provide links for additional information on each workflow. 

**Describe alternatives you've considered**
This can be added in a new section. 

**Additional context**
Reference: https://docs.github.com/en/actions/quickstart
Example: https://github.com/TommyLin/focus-stacking/wiki/Github-Actions

**Requester Info**
Ariel Adams, ASRC Federal

### PR DESCRIPTION
**Pull request by ArielSAdamsNASA**
**Checklist (Please check before submitting)**

* [x] I reviewed the [Contributing Guide](https://github.com/nasa/cFS/blob/main/CONTRIBUTING.md).
* [x] I signed and emailed the appropriate Contributor License agreement to GSFC-SoftwareRelease@mail.nasa.gov and copied cfs-program@lists.nasa.gov.

**Describe the contribution**
Fixes #240
- Added instructions on how to view errors from failed workflows
- Added instructions on how to view workflow results in the Actions tab and in pull requests
- Added a list of our workflows and the status of them
- Added information on how to configure workflows and where to find them

**Expected behavior changes**
Users should know how to use our workflows.

**Additional context**
Used Reference: https://docs.github.com/en/actions/quickstart
Used Example: https://github.com/TommyLin/focus-stacking/wiki/Github-Actions

**Contributor Info - All information REQUIRED for consideration of pull request**
Ariel Adams, ASRC Federal

### CODE REVIEW

**Review by skliper — APPROVED** [review]

**Comment by ArielSAdamsNASA** [comment]
Added status badge to [OSAL's local unit test workflow.](https://github.com/nasa/osal/actions/workflows/local_unit_test.yml)

**Comment by skliper** [comment]
CCB:2021-04-14 - APPROVED

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -109,12 +109,68 @@ Follow GitHub's fork-branch-pull request pattern.
...
+
+or 
+
+1. In pull requests, click the Checks tab. 
+2. From the list of workflow runs, click the name of the run you want to see.
+4. Scroll to the bottom of the page and download the artifacts. For CodeQL results, expand Code scanning alerts at the bottom of the list of checks and select CodeQL. 
+
+#### Configure Workflows
+
+1. Workflows are under [.github/workflows](https://github.com/nasa/cFS/tree/main/). CodeQL uses a .yml file and a configuration file under .github/codeql(https://github.com/nasa/cFS/tree/main/.github/codeql).
```
Comment (astrogeco) [inline CONTRIBUTING.md:146]: Fix broken link syntax

Comment (astrogeco) [reply CONTRIBUTING.md:146]: Move Code QL info to Code QL Analysis entry in the "our workflows" secton

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -109,12 +109,68 @@ Follow GitHub's fork-branch-pull request pattern.
...
 4. For any changes to existing API's, alter the unit tests to cover the changes (and remove tests made irrelevant due to your changes).
 
-5. Review the static code analyses results. 
+5. Review the static code analyses results from the Static Analysis and CodeQL Analysis workflows. For more information on how to use these workflows, view [Using GitHub Actions Workflows](#using-github-actions-workflows).  
     - Push code changes to the appropriate forked repository. 
     - Go to the Actions tab and enable GitHub Actions Workflow. The CodeQL Analyis and Static Analysis will be triggered for all pushed code. 
     - Review these workflows for any warnings or errors. 
     - Once successful, create a pull request. 
 
+### Using GitHub Actions Workflows
```
Comment (astrogeco) [inline CONTRIBUTING.md:118]: Maybe add something about how these can be seen in their forks?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -99,7 +99,7 @@ Follow GitHub's fork-branch-pull request pattern.
 ## Writing High-Quality Code
 
 1. Follow cFS code conventions (formatting, symbol naming, file naming, etc). Do *not* change/reformat existing code, except to address your changes.
-    - The cFS submodules uses the Format Check workflow to ensure users follow the [clang-format-10](https://releases.llvm.org/10.0.0/tools/clang/docs/ClangFormat.html) style. 
+    - The cFS submodules uses the Format Check workflow to ensure users follow the [clang-format-10](https://releases.llvm.org/10.0.0/tools/clang/docs/ClangFormat.html) style. For more information on how to use the Format Check workflow, view [Using GitHub Actions Workflows](#using-github-actions-workflows).
```
Comment (astrogeco) [inline CONTRIBUTING.md:102]: I would point to the .clang-format file under instead of the clang format tool repo.

**Review by astrogeco — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (ArielSAdamsNASA) [reply CONTRIBUTING.md]: Fixed link and moved CodeQL information to the CodeQL Analysis section.

File: CONTRIBUTING.md (hunk 2 above)
Comment (ArielSAdamsNASA) [reply CONTRIBUTING.md:118]: Fixed. 
> All of our workflows will be available for forked repositories once enabled. To enable workflows, navigate to the Actions tab and click "I understand my workflows, go ahead and enable them".

File: CONTRIBUTING.md (hunk 3 above)
Comment (ArielSAdamsNASA) [reply CONTRIBUTING.md]: Changed link from https://releases.llvm.org/10.0.0/tools/clang/docs/ClangFormat.html to https://github.com/nasa/cFS/blob/main/.clang-format.

**Review by astrogeco — APPROVED** [review]

**Pull request merged**